pkg/kubeconfig/kubeconfig.go

```diff
@@ -25,15 +29,26 @@ func FromEnvironment() string {
 	return os.Getenv(EnvName)
 }
 
-type missingFileError struct {
-	Path string
-}
+// ValidateFile loads a file to validate it's basic contents.
+//
+// The values of the fields within aren't validated, but the file's existence
+// and basic structure are checked.
+func ValidateFile(filename string) error {
```

Comment:
Do we want to simplify this to just `Validate`?

Reply:
What can I say, I want to keep the hope alive for a reader based implementation some day. :)